### PR TITLE
Fix ContainsConstraint ignoring IgnoreCase for collection

### LINKED ISSUE #3039: ContainsConstraint ignores IgnoreCase in case of actual collection argument
```
var items = new[]{"ABC", "dce"};
Assert.That(items, Does.Contain("abc").IgnoreCase);
```

Such assertion fails with message:
  Expected: some item equal to "abc"
  But was:  < "ABC", "dce" >

### PR DESCRIPTION
**Pull request by Dreamescaper**
fixes #3039

### CODE REVIEW

File: src/NUnitFramework/framework/Constraints/ContainsConstraint.cs [hunk 1]
```diff
@@ -53,9 +53,12 @@ public ContainsConstraint(object expected)
...
         public override string Description
         {
-            get { return _realConstraint != null ? 
-                    _realConstraint.Description : 
-                    "containing " + MsgUtils.FormatValue(_expected); }
+            get
+            {
+                return _realConstraint != null ?
+                  _realConstraint.Description :
+                  "containing " + MsgUtils.FormatValue(_expected);
```
[the remark this answers is not in the document]
Comment (ChrisMaddock) [reply src/NUnitFramework/framework/Constraints/ContainsConstraint.cs]: https://github.com/nunit/nunit/blob/cefc0c3a86c065daa4130aa255b1384db51b2b08/src/NUnitFramework/framework/Constraints/StringConstraint.cs#L55-L61

Here's an example of what I meant, although the 'ignore case' bit might actually be added by the equal constraint in this case? I'm not sure how it works off the top of my head - could you mind checking the result is coming out right in the end, however? (And maybe a unit test, to cover that too? 🙂)

File: src/NUnitFramework/tests/Constraints/ContainsConstraintTests.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        [Test]
+        public void HonorsIgnoreCaseForString()
+        {
+            var actualString = "ABCdef";
+            var constraint = new ContainsConstraint("abc").IgnoreCase;
+
+            var result = constraint.ApplyTo(actualString);
+
+            Assert.That(result.IsSuccess);
+        }
```
[the remark this answers is not in the document]
Comment (Dreamescaper) [reply src/NUnitFramework/tests/Constraints/ContainsConstraintTests.cs]: There are classes ContainsTest and ContainsTest_IgnoreCase in Syntax.StringConstraints.

Do you think anything else needed?

File: src/NUnitFramework/framework/Constraints/ContainsConstraint.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (Dreamescaper) [reply src/NUnitFramework/framework/Constraints/ContainsConstraint.cs]: yeah, in most cases it should take either someItem or substring constraint description. Updated with 'ignore case' details though, will make no harm.

Added tests for error messages as well

**Comment by Dreamescaper** [comment]
Travis failure seems to be unrelated

File: src/NUnitFramework/tests/Constraints/ContainsConstraintTests.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (ChrisMaddock) [reply src/NUnitFramework/tests/Constraints/ContainsConstraintTests.cs]: Ah ok! Yep, looks good - thanks for checking!

**Pull request merged**